stk/backend/triton_kernels.py

```diff
@@ -0,0 +1,374 @@
+    rm =  pid_m * BLOCK_M + tl.arange(0, BLOCK_M)
+    rak = tl.arange(0, BLOCK_K)
+
+    A += (rm[:, None] * stride_am + rak[None, :] * stride_ak)
+
+    # pointers to sparse matrix
+    rn = tl.arange(0, BLOCK_N)
+    rbk = tl.arange(0, BLOCK_K)
+    B += (rbk[:, None] * stride_bk + rn[None, :] * stride_bn)
+
+    # do matrix multiplication
+    acc = tl.zeros((BLOCK_M, BLOCK_N), dtype=ACC_TYPE)
+    nsub_blocks = tl.cdiv(BLOCK_SIZE, BLOCK_K)
+
+    BLOCK_ELEMENTS = BLOCK_SIZE * BLOCK_SIZE
+    
+    ak_sub_incr = BLOCK_K * stride_ak
+    ak_block_incr = BLOCK_SIZE * stride_ak
+    bk_sub_incr = BLOCK_K * stride_bk
+
+    for k in range(nsub_blocks * (end_inx - start_inx)):
+        sub_block_inx = k % nsub_blocks
+        block_inx = k // nsub_blocks
+
+        if trans_B:
+            ptr_B = B + (start_inx + block_inx) * BLOCK_ELEMENTS + sub_block_inx * bk_sub_incr
+        else:
+            ptr_B = B + tl.load(block_offsets_t + start_inx + block_inx) * BLOCK_ELEMENTS + sub_block_inx * bk_sub_incr
+
+        ptr_A = A + tl.load(column_indices + start_inx + block_inx) * ak_block_incr + sub_block_inx * ak_sub_incr
+        a = tl.load(ptr_A)
+        b = tl.load(ptr_B)
+        acc += tl.dot(a, b) 
+
+    acc = acc.to(C.dtype.element_ty)
+    cm = pid_m * BLOCK_M + tl.arange(0, BLOCK_M)
+    cn = pid_n * BLOCK_N + tl.arange(0, BLOCK_N)
+    C = C + (cm[:, None] * stride_cm + cn[None, :] * stride_cn)
+    tl.store(C, acc, mask=True)
+
+def dsd(shape,
+        data,
+        offsets,
+        row_indices,
+        column_indices,
+        offsets_t,
+        column_indices_t,
+        block_offsets_t,
+        transpose_a,
+        rhs,
+        out
+    ):
+
+    device = rhs.device
+    trans_A = transpose_a
+    trans_B = False
+
+    if rhs.stride(0) > 1 and rhs.stride(1) > 1:
+        trans_B = True
+
+    # checks constraints
+    assert shape[1] == rhs.shape[0], "incompatible dimensions"
+    M, K = shape
+    _, N = rhs.shape
+    
+    # accumulator types
+    ACC_TYPE = tl.float32 if rhs.dtype in [torch.float16, torch.bfloat16, torch.float32] else tl.int32
+
+    stride_am, stride_ak = data.stride(1), data.stride(2)
+    stride_bk, stride_bn = rhs.stride(0), rhs.stride(1)
+    a_column_indices  = column_indices
+    a_offsets = offsets
+
+    # launch kernel
+    grid = lambda META: (triton.cdiv(M, META['BLOCK_M']), triton.cdiv(N, META['BLOCK_N']))
+
+    if trans_A:
+        stride_am, stride_ak = data.stride(2), data.stride(1)
+        a_column_indices, a_offsets = column_indices_t, offsets_t
+
+    if trans_B:
+        stride_bk, stride_bn = rhs.stride(1), rhs.stride(0) 
+
+    _dsd_kernel[grid](
+        data.data, rhs, out, M, N, K,
+        stride_am, stride_ak,
+        stride_bk, stride_bn,
+        out.stride(0), out.stride(1),
+        row_indices, a_column_indices, a_offsets,
+        block_offsets_t, trans_A, trans_B,
+        GROUP_M=128, ACC_TYPE=ACC_TYPE
+    )
+    # return out
+
+def dds(lhs,
```

Comment:
If you remove DDS above for a version of DSD that supports C transposition I think you can remove this as well.